### PR TITLE
Wr/twinning

### PR DESCRIPTION
**Pull request by WillieRuemmele**
### What does this PR do?
Adds `--dev-debug` flag with enhancements
You can now `--dev-debug sf:config` to limit DEBUG statements
✅ : the current `promote-verify` uses the correct command

### Acceptance Criteria
run a command with `--dev-debug` to see the debug output, should be equivalent to setting `DEBUG=*` before running a command

### Testing Notes
`--dev-debug` results in `*` debug statements
`--dev-debug <filter>` will filter the debug statements

### Checklist
- [x] This change has been approved by the CLI Review Board or approval isn't required. Anything that adds/changes/deletes commands, flags, output, or json output has to be reviewed.
- [x] Does this follow the [deprecation policy](https://developer.salesforce.com/docs/atlas.en-us.sfdx_dev.meta/sfdx_dev/sfdx_dev_cli_deprecation.htm)?

### What issues does this PR fix or reference?
@W-12152987@

### CODE REVIEW

File: src/flags.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        let debug = '*';
+        if (process.argv[index + 1] && !process.argv[index + 1]?.startsWith('-')) {
+          // if the next arg is defined and IS NOT a flag, set it as the logger level
+          debug = process.argv[index + 1];
+        }
+
+        // convert --dev-debug into a set of environment variables
+        process.env.DEBUG = debug;
+        process.env.SF_DEBUG = '1';
+        process.env.SF_ENV = 'development';
```
Comment (cristiand391) [inline src/flags.ts:32]: are these env var documented?

I don't see any ref to `SF_DEBUG`:
https://github.com/search?q=SF_DEBUG+org%3Asalesforcecli+&type=code

for `SF_ENV` only this:
https://github.com/search?q=SF_ENV%20org%3Asalesforcecli%20&type=code

Comment (WillieRuemmele) [reply src/flags.ts:32]: I converted them from `SFDX_` to `SF_`, but didn't think to use that global GH search. 

I don't see useful references for `SFDX_DEBUG`, 

There's a few more hits for `SFDX_ENV` : https://github.com/search?q=SFDX_ENV+org%3Asalesforcecli+&type=code

but I'm not sure we need either of them 🤔

**Review by mshanemc — DISMISSED** [review]

**Comment by mshanemc** [comment]
QA notes

`./bin/run env list --dev-debug` 
✅ does all the debug output as expected
✅ ran with a few flags (--sort with a value, --help) after --dev-debug
✅ ran with a non-`*` debug value (`sf:core`) and saw only those logs

example from the comments
`config get --dev-debug target-dev-hub --json` 
❌ command fails with `NoConfigKeysFoundError` (code is deleting the required argument after `--dev-debug`)
💡 what if this code ensured that, if present, --dev-debug is the last arg (or penultimate arg if followed by a non-flag value) ?  I'm not sure there's a good way to tell an arg from a debug value

File: src/flags.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+export interface ProcessLike {
+  argv: string[];
+  env: { [key: string]: string | undefined };
+}
+
+export function preprocessCliFlags(process: ProcessLike): void {
+  process.argv.map((arg) => {
+    if (arg === '--dev-debug') {
+      let debug = '*';
+      const filterIndex = process.argv.indexOf('--dev-filter');
```
Comment (mshanemc) [inline src/flags.ts:23]: I don't like that name.  Is it a `--log-filter` or a `--debug-filter`.  It's not filtering devs.

File: src/flags.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      if (filterIndex > 0) {
+        debug = process.argv[filterIndex + 1];
+
+        process.argv.splice(filterIndex + 1, 1);
+        process.argv.splice(filterIndex, 1);
```
Comment (mshanemc) [inline src/flags.ts:28]: is that the same as `process.argv.splice(filterIndex, 2)`

**Review by mshanemc — APPROVED** [review]

**Review by mshanemc — APPROVED** [review]

**Comment by mshanemc** [comment]
QA notes:

previous issue fixed.  Tried all of
./bin/run config get --debug-filter sf:core target-dev-hub --dev-debug
./bin/run config get --debug-filter sf:core target-dev-hub         
./bin/run config get --dev-debug --debug-filter sf:core target-dev-hub
./bin/run config get --dev-debug --debug-filter  target-dev-hub     

All passed/failed as expected.

**Pull request merged**